var/spack/repos/builtin/packages/py-numpy/package.py

```diff
@@ -332,6 +379,34 @@ def setup_build_environment(self, env):
             lapack = 'lapack'
 
         env.set('NPY_LAPACK_ORDER', lapack)
+        # https://github.com/scipy/scipy/issues/9080
+        env.set('F90', spack_fc)
+
+    def fortran_args(self, spec, prefix):
+        args = []
+        if spec.satisfies('%fj'):
+            args.extend(['config_fc', '--fcompiler=fj'])
```

Comment:
I remember seeing something like this for Intel too: https://software.intel.com/content/www/us/en/develop/articles/build-numpy-with-mkl-and-icc.html